### PR TITLE
feat: refactor external api providers and code warnings

### LINKED ISSUE #316: Refactor External API providers and code warnings
Refactor Oracle API and ForestClient API calls to have a single place where the calls are made from. Do not duplicate code or logic.

### PR DESCRIPTION
**Pull request by RMCampos**
# Description

- Remove all unnecessary `@Autowired`;
- Create a single `Provider` class to put all retrieving code in there and extend new providers from it;
- Create a ProviderConfig to put all api and related configurations in one place;

Fixes #316 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

These changes were tested locally with Swagger, Unit testing and Integration tests.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

### CODE REVIEW

File: backend/src/main/java/ca/bc/gov/backendstartapi/service/ForestClientService.java [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
 import org.springframework.web.client.HttpClientErrorException;
 
 /** Service to deal with {@link ForestClientDto ForestClients}. */
 @Service
-@RequiredArgsConstructor
 @Slf4j
 public class ForestClientService {
 
-  private final ForestClientApiProvider provider;
+  private ForestClientApiProvider provider;
```
Comment (paulushcgcj) [inline backend/src/main/java/ca/bc/gov/backendstartapi/service/ForestClientService.java:15]: You can use SOLID principles and depend on the generic, and use `@Qualifier` to serve the correct one here

**Review by paulushcgcj — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ca/bc/gov/backendstartapi/provider/Provider.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+   * @param apiUrl The API URL address.
+   * @param uriVars Parameters and values for the API address (optional). You can call the method
+   *     createParamsMap to create those.
+   * @return A single object of the given className.
+   */
+  protected <T2> T2 doGetRequestSingleObject(
+      Class<T2> className, String apiUrl, Map<String, String> uriVars) {
+    HttpEntity<Void> requesEntity = getRequestEntity();
+
+    logParams(uriVars);
```
[the remark this answers is not in the document]
Comment (RMCampos) [reply backend/src/main/java/ca/bc/gov/backendstartapi/provider/Provider.java:44]: Sorry. This what? I don't follow. @paulushcgcj

[the remark this answers is not in the document]
Comment (RMCampos) [reply backend/src/main/java/ca/bc/gov/backendstartapi/provider/Provider.java]: I ended up changind all of this code. Now there's an Interface responsible for managing the providers.

File: backend/src/main/java/ca/bc/gov/backendstartapi/service/ForestClientService.java (hunk 1 above)
Comment (RMCampos) [reply backend/src/main/java/ca/bc/gov/backendstartapi/service/ForestClientService.java]: Yup. [👍🏻 Done here](https://github.com/bcgov/nr-spar/pull/317/commits/c27d7486f76fe34426d5968fe2f229e1c75641d1)

**Comment by RMCampos** [comment]
I tested all related endpoints, everything is working!

- ForestClient API
- Oracle API
- Others endpoints

**Review by craigyu — APPROVED** [review]
💯 Great changes, reviewed together in a meeting

**Pull request merged**